dotCMS/src/main/webapp/html/portlet/ext/contentlet/contentlet_actions_inc.jsp

```diff
@@ -127,11 +147,10 @@ function jumpToContentType(){
 <%}%>
 
 <%--check permissions to display the save and publish button or not--%>
-<%boolean canUserWriteToContentlet = conPerAPI.doesUserHavePermission(contentlet,PermissionAPI.PERMISSION_WRITE,user, PageMode.get(request).respectAnonPerms);%>
 
-<%if(!"edit-page".equals(request.getParameter("angularCurrentPortlet")) && contentlet.isHTMLPage() && contentlet.getIdentifier() != "" && (canUserWriteToContentlet)) {%>
+<%if(!"edit-page".equals(request.getParameter("angularCurrentPortlet")) && UtilMethods.isSet(previewUrl)) {%>
    <div class="content-edit-actions" >
-       <a style="border:0px;" onClick="editPage('<%= APILocator.getIdentifierAPI().find(contentlet.getIdentifier()).getURI() %>', '<%= contentlet.getLanguageId() %>')">
+       <a style="border:0px;" href="<%= previewUrl %>" target="_blank">
```

Comment:
When you are creating a page from content search the buttons shows and if you click it it throws a 404, we should only show the button when the page actually exist.

Reply:
Noted

Reply:
updated